api/apps/articles/model/retrieve.coffee

```diff
@@ -1,88 +1,101 @@
 _ = require 'underscore'
-Joi = require 'joi'
-Joi.objectId = require('joi-objectid') Joi
-schema = require './schema'
-{ ObjectId } = require 'mongojs'
 moment = require 'moment'
 
 @toQuery = (input, callback) ->
-  Joi.validate input, schema.querySchema, { stripUnknown: true }, (err, input) ->
-    return callback err if err
-    # Separate "find" query from sort/offest/limit
-    { limit, offset, sort, count } = input
-    query = _.omit input, 'limit', 'offset', 'sort', 'artist_id', 'artwork_id',
-      'super_article_for', 'fair_ids', 'fair_programming_id', 'fair_artsy_id',
-      'fair_about_id', 'partner_id', 'auction_id', 'show_id', 'q',
-      'all_by_author', 'section_id', 'tags', 'tracking_tags', 'has_video', 'fair_id',
-      'channel_id', 'ids', 'scheduled', 'count'
-    # Type cast IDs
-    # TODO: https://github.com/pebble/joi-objectid/issues/2#issuecomment-75189638
-    query.author_id = ObjectId input.author_id if input.author_id
-    if input.fair_ids
-      query.fair_ids = { $elemMatch: { $in: _.map(input.fair_ids, ObjectId) } }
-    query.fair_ids = ObjectId input.fair_id if input.fair_id
-    if input.fair_programming_id
-      query.fair_programming_ids = ObjectId input.fair_programming_id
-    query.fair_artsy_ids = ObjectId input.fair_artsy_id if input.fair_artsy_id
-    query.fair_about_ids = ObjectId input.fair_about_id if input.fair_about_id
-    query.partner_ids = ObjectId input.partner_id if input.partner_id
-    query.show_ids = ObjectId input.show_id if input.show_id
-    query.auction_ids = ObjectId input.auction_id if input.auction_id
-    query.section_ids = ObjectId input.section_id if input.section_id
-    if input.biography_for_artist_id
-      query.biography_for_artist_id = ObjectId input.biography_for_artist_id
-    query.featured_artwork_ids = ObjectId input.artwork_id if input.artwork_id
-    query.tags = { $in: input.tags } if input.tags
-    query.tracking_tags = { $in: input.tracking_tags } if input.tracking_tags
+  { limit, offset, sort, count } = input
+  query = _.omit input,
+    'limit'
```

Comment:
Oh yeah, I'm kinda a fanatic about alpha 🙂 Everything I come across I tend to sort